File: LEETCODE/LEETCODE-CPP/383.cpp
//383. Ransom Note
//https://leetcode.com/problems/ransom-note/

class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        
        unordered_map<char,int>m1;
        
        for(int i=0;i<magazine.size();i++)
            m1[magazine[i]]++;
        
        for(auto i: ransomNote)
            if(!m1[i])
                return false;
            
            else
                m1[i]--;
        
        return true;
        
    }
};

File: LEETCODE/LEETCODE-PYTHON/383.py
# 383. Ransom Note
# https://leetcode.com/problems/ransom-note/description/

# Method -1 brute force
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        freq=defaultdict(int)

        for i in magazine:
            freq[i]+=1
        
        for x in ransomNote:
            if x in freq and freq[x]!=0:
                freq[x]-=1
            else:
                return False
        
        return True
        